src/main/java/subway/line/LineService.java

```diff
@@ -0,0 +1,75 @@
+package subway.line;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.transaction.Transactional;
+
+import org.springframework.stereotype.Service;
+
+import subway.Station;
+import subway.StationRepository;
+import subway.StationResponse;
+
+@Service
+public class LineService {
+	private LineRepository lineRepository;
+	private StationRepository stationRepository;
+
+	public LineService(LineRepository lineRepository, StationRepository stationRepository) {
+		this.lineRepository = lineRepository;
+		this.stationRepository = stationRepository;
+	}
+
+	@Transactional
+	public LineResponse saveLine(LineRequest lineRequest) {
+		Station upStation = stationRepository.findById(lineRequest.getUpStationsId()).orElse(null);
+		Station downStation = stationRepository.findById(lineRequest.getDownStationsId()).orElse(null);
+
+		Line saveLine = lineRepository.save(new Line(
+			lineRequest.getName(), lineRequest.getColor(),
+			upStation, downStation, lineRequest.getDistance()));
+
+		return createLineResponse(saveLine);
+	}
+
+	public List<LineResponse> findAllLines() {
```

Comment:
```suggestion
    @Transactional(readonly=true)
	public List<LineResponse> findAllLines() {
```

`@Transactional(readonly=true)` 를 사용해볼까요? readonly를 true의 장점은 무엇 일까요?

Reply:
읽기만을 사용한다는 명시적인 표현이 될 수 있고,
변경감지를 사용하지 않는 것을 보면 영속성 컨텍스트의 관리를 받지 않아 조금 더 빠를 것 같습니다.